### PR TITLE
obs-webrtc: Improve WHIP compliance

### PR DESCRIPTION
**Pull request by Sean-Der**
### Description
Improves WHIP compliance around the handling of Location headers.

If the location header isn't present we don't allow the session to start. We also now support Absolute and Relative Location headers. 

This cherry-picks https://github.com/obsproject/obs-studio/pull/9306 and then adds additional checks

### Motivation and Context
I want to make sure we are as compliant as possible. If we ship a release with compliance issues we will have to support it forever. 

### How Has This Been Tested?
Tested against 
* AWS IVS
* Broadcast Box
* Cloudflare

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Sean-Der** [comment]
@notr1ch This covers all the compliance issues you raised in https://github.com/obsproject/obs-studio/pull/9306 I believe!

**Comment by RytoEX** [comment]
This appears to contain #9306, which should be noted in the PR description.

**Comment by Sean-Der** [comment]
@RytoEX updated the description. Does that read ok?

**Comment by RytoEX** [comment]
> @RytoEX updated the description. Does that read ok?

Yes, that's fine.

File: plugins/obs-webrtc/whip-output.cpp
Comment (DDRBoxman) [inline plugins/obs-webrtc/whip-output.cpp:366]: TIL http finally supports relative redirects in Location headers, neat.

Can we stick a comment here saying that this is what were doing here?

Also should we check that it's a valid url path too or is the protocol check good enough?

File: plugins/obs-webrtc/whip-output.cpp
Comment (DDRBoxman) [inline plugins/obs-webrtc/whip-output.cpp:363]: Nit-picky but can we give this a more descriptive name? my head keeps reading the h as header further down, heh

File: plugins/obs-webrtc/whip-output.cpp
Comment (Sean-Der) [reply plugins/obs-webrtc/whip-output.cpp:368]: Done! Added a comment on Relative vs Absolute URL handling

File: plugins/obs-webrtc/whip-output.cpp
Comment (Sean-Der) [reply plugins/obs-webrtc/whip-output.cpp]: Done! Changed name to `url_builder` I think that better communicates its purpose.

**Comment by Sean-Der** [comment]
@DDRBoxman thank you for the review! I addressed both comments, mind looking again?

**Review by DDRBoxman — APPROVED** [review]

**Comment by RytoEX** [comment]
To those looking to merge: I would like to see #9306 approved and merged first if this is meant to stack on top of it. If this is meant to supersede #9306 entirely, then I would prefer that it is closed first.

**Comment by Sean-Der** [comment]
@RytoEX This is meant to supersede. Can you please #9306 ? This PR modifies the commit that was cherry-picked

File: plugins/obs-webrtc/whip-output.cpp
Comment (notr1ch) [inline plugins/obs-webrtc/whip-output.cpp:356]: This should probably be `LOG_ERROR` now.

File: plugins/obs-webrtc/whip-output.cpp
Comment (notr1ch) [inline plugins/obs-webrtc/whip-output.cpp:393]: Also here.

File: plugins/obs-webrtc/whip-output.cpp
Comment (Sean-Der) [reply plugins/obs-webrtc/whip-output.cpp]: Changed to `LOG_ERROR`

File: plugins/obs-webrtc/whip-output.cpp
Comment (Sean-Der) [reply plugins/obs-webrtc/whip-output.cpp]: Changed to `LOG_ERROR`

**Comment by Sean-Der** [comment]
@notr1ch I have updated all the `LOG_WARNING` messages appropriately. In cases where streaming wouldn't start because of an issue I have changed them.

Can I get another review please? thank you

**Comment by Sean-Der** [comment]
@RytoEX Yes can you please close it, thank you!

**Pull request merged**
